src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,215 @@
+            process = Process.Start(processInfo);
+        }
+
+        //Grab the JSON from the github API and deserialize
+        private static async Task<GHUpdate> GetGithubUpdates()
+        {
+            HttpClient client = new HttpClient();
+
+            var request = new HttpRequestMessage()
+            {
+                RequestUri = new Uri("https://api.github.com/repos/momentum-mod/lumper/releases"),
+                Method = HttpMethod.Get
+            };
+            client.DefaultRequestHeaders.Add("User-Agent", "Other");
+            HttpResponseMessage m = client.Send(request);
+
+            if (m.IsSuccessStatusCode)
+            {
+                string response = await m.Content.ReadAsStringAsync();
+
+                var assets = JsonConvert.DeserializeObject<GHUpdate>(JsonObject.Parse(response)[0].ToString());
+                return assets;
+            }
+            else
+                throw new Exception("Could not connect - error " + m.StatusCode);
+        }
+        
+        public static async Task<bool> CheckForUpdate()
+        {
+            GHUpdate assets = await GetGithubUpdates();
+
+            //parse tag name to find the current and latest version
+            //finding the format of xx.yy.zz
+            string newestVersionSplit = Regex.Match(assets.tag_name, "[0-9]+\\.[0-9]+\\.[0-9]+").ToString();
+            MMP current = GetMMPVersion(Assembly.GetExecutingAssembly().GetName().Version.ToString());
+            MMP latest = GetMMPVersion(newestVersionSplit);
+
+            return (current.major != latest.major || current.minor != latest.minor || current.patch != latest.patch);
+        }
+        //returns the URL to the download link for the OS-specific version
+        private static string GetPath(GHUpdate assets, string OS)
+        {
+            for (int i = 0; i < assets.assets.Length; i++)
+            {
+                if (assets.assets[i].name.ToLower().Contains(OS.ToLower()))
+                {
+                    return assets.assets[i].browser_download_url;
+                }
+
+            }
+            return null;
+        }
+
+        public static async ValueTask Update()
+        {
+
+            GHUpdate assets = await GetGithubUpdates();
+
+            string newestVersionSplit = Regex.Match(assets.tag_name, "[0-9]+\\.[0-9]+\\.[0-9]+").ToString();
+            MMP latest = GetMMPVersion(newestVersionSplit);
+
+
+            //NOTE: linux is untested
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                string fileURL = GetPath(assets, "linux");
+                string fileName = "linux_" + latest.major + "." + latest.minor + "." + latest.patch + ".zip";
+                string directoryName = fileName + "temp";
+
+                //download and unzip to a temp directory
+                WebClient webClient = new WebClient();
+                webClient.DownloadFile(new Uri(fileURL), fileName);

```

Comment:
Use HttpClient not WebClient, they both do same stuff, HttpClient is just newer.